src/org/intellij/erlang/psi/impl/ErlangPsiImplUtil.java

```diff
@@ -1821,6 +1829,48 @@ public static boolean fromTheSameCaseExpression(@NotNull PsiElement origin, @Not
     return false;
   }
 
+  public static class ErlangFunctionCallNamedParameter<T extends PsiElement> extends PatternCondition<T> {
+    private final List<String> NAMES;
+
+    public ErlangFunctionCallNamedParameter(@NotNull String... names) {
+      super("functionCallNamedParameter");
+      NAMES = ContainerUtil.newArrayList(names);
+    }
+
+    @Override
+    public boolean accepts(@NotNull T element, ProcessingContext context) {
+      ErlangQAtom argument = element instanceof ErlangQAtom ? (ErlangQAtom) element : null;
+      if (argument == null) return false;
+
+      ErlangFunctionCallExpression call = PsiTreeUtil.getParentOfType(argument, ErlangFunctionCallExpression.class);
```

Comment:
You're looking for function call expression recursively - it's both wrong, and bad performance-wise.